### PR TITLE
FEAT: torch.compile on policy models for potential speed-ups. 

### PR DESCRIPTION
**Pull request by Pythoniasm**
FEAT: torch.compile on policy models for potential speed-ups.
Also updated torch version to torch==2.0.0 and Python version to Python 3.10

## Motivation and Context

- [ ] I have raised an issue to propose this change ([required](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) for new features and bug fixes)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)

## Checklist

- [ ] I've read the [CONTRIBUTION](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) guide (**required**)
- [ ] I have updated the changelog accordingly (**required**).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [ ] I have updated the documentation accordingly.
- [ ] I have opened an associated PR on the [SB3-Contrib repository](https://github.com/Stable-Baselines-Team/stable-baselines3-contrib) (if necessary)
- [ ] I have opened an associated PR on the [RL-Zoo3 repository](https://github.com/DLR-RM/rl-baselines3-zoo) (if necessary)
- [ ] I have reformatted the code using `make format` (**required**)
- [ ] I have checked the codestyle using `make check-codestyle` and `make lint` (**required**)
- [ ] I have ensured `make pytest` and `make type` both pass. (**required**)
- [ ] I have checked that the documentation builds using `make doc` (**required**)

Note: You can run most of the checks using `make commit-checks`.

Note: we are using a maximum length of 127 characters per line

### CODE REVIEW

**Comment by araffin** [comment]
Hello,
thanks for the PR.
As you may have noticed, we don't actually need to modify SB3 internal in order to use torch compile (we should probably update the doc):
```python
import torch as th

model.policy = th.compile(model.policy, mode="max-autotune")
```
Unfortunately, the current bottleneck is in the gradient update (so the compile will probably a small speed up but not too massive if there are many gradient updates per step).
If you want to speed up the gradient update, you may have a look at https://github.com/araffin/sbx


(btw, probably mode should be either default or "reduce-overhead").

**Comment by araffin** [comment]
Closing in favor of https://github.com/DLR-RM/stable-baselines3/pull/1439
